File: shared/src/main/java/ru/privatenull/pnauth/dialog/DialogLayout.java
package ru.privatenull.pnauth.dialog;

import net.kyori.adventure.text.Component;
import java.util.List;

/** Fields shared by every vanilla dialog type. */
public record DialogLayout(Component title, Component externalTitle, List<DialogBody> body,
                           List<DialogInput> inputs, boolean canCloseWithEscape, boolean pause,
                           AfterAction afterAction) {
    public DialogLayout {
        externalTitle = externalTitle == null ? title : externalTitle;
        body = List.copyOf(body);
        inputs = List.copyOf(inputs);
        afterAction = afterAction == null ? AfterAction.CLOSE : afterAction;
    }

    public enum AfterAction { CLOSE, NONE, WAIT_FOR_RESPONSE }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/dialog/DialogLayout.kt
package ru.privatenull.pnauth.dialog

import net.kyori.adventure.text.Component

/** Fields shared by every vanilla dialog type. */
@JvmRecord
data class DialogLayout @JvmOverloads constructor(
    val title: Component,
    val externalTitle: Component? = title,
    val body: List<DialogBody> = emptyList(),
    val inputs: List<DialogInput> = emptyList(),
    val canCloseWithEscape: Boolean = true,
    val pause: Boolean = false,
    val afterAction: AfterAction = AfterAction.CLOSE
) {
    enum class AfterAction { CLOSE, NONE, WAIT_FOR_RESPONSE }
}
